x/programs/examples/storage/storage.go

```diff
@@ -17,6 +17,8 @@ const (
 	programPrefix = 0x0
 )
 
+// ProgramPrefixKey returns a properly formatted key
+// for storing a value at [id][key].
 func ProgramPrefixKey(id []byte, key []byte) (k []byte) {
```

Comment:
This function is wrong. You are setting byte 0 then overriding it with an id copy.